### PR TITLE
Improvement/combine common table functionally

### PR DESCRIPTION
**Pull request by JWZeta**

### CODE REVIEW

File: include/server_table.h [hunk 1]
```diff
@@ -16,50 +16,6 @@
 #include <openssl/err.h>
```
Comment (b-carr) [inline include/server_table.h:16]: Do we anticipate this file containing anything in the immediate future? If not we should probably just get rid of it.

File: sgx/pelz_enclave.edl [hunk 2]
```diff
@@ -4,77 +4,41 @@ enclave {
...
+ * This function destroys the table specified by type.
  * <pre>
  *
- * @param[in] key_table The key table to be destroyed
+ * @param[in] type The type of table to be destroyed (ex: key_table or server_table)
  *
  * @return 0 on success, 1 on error
  */
-public int key_table_destroy(void);
+public int table_destroy(int type);
```
Comment (b-carr) [inline sgx/pelz_enclave.edl:24]: The argument for this is really a ```TableType```.

File: sgx/pelz_enclave.edl [hunk 3]
```diff
@@ -4,77 +4,41 @@ enclave {
...
- * <pre>
- * This function to delete values in hash table based on location in key_id.
- * <pre>
- *
- * @param[in] server_id.chars Server identifier assumed to be null terminated
- * @param[in] server_id.len The length of the server identifier
- * @param[in] server_table The server table that the cert needs to be deleted from
- * @param[out] server_table The server table without the deleted cert
+ * @param[in] type The table type that the id and value needs to be deleted from
+ * @param[in] id.chars Identifier assumed to be null terminated
```
Comment (b-carr) [inline sgx/pelz_enclave.edl:32]: If we pass in the identifier length why do we bother assuming the identifier is null terminated?

File: sgx/pelz_enclave.edl [hunk 4]
```diff
@@ -4,77 +4,41 @@ enclave {
...
+ * @param[in] id.len The length of the identifier
  *
  * @return 0 on success, 1 on error
  */
-public int server_table_delete(charbuf server_id);
+public int table_delete(int type, charbuf id);
```
Comment (b-carr) [inline sgx/pelz_enclave.edl:37]: Correct the type of the ```type``` argument.

File: src/util/common_table.c [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+};
+
+Table server_table = {
+  .entries = NULL,
+  .num_entries = 0,
+  .mem_size = 0
+};
+
+//Destroy server table
+int table_destroy(int type)
```
Comment (b-carr) [inline src/util/common_table.c:33]: Correct the type of ```type```

File: src/util/common_table.c [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+  //Free the storage allocated for the hash table
+  free(table.entries);
+  table.entries = NULL;
+  table.num_entries = 0;
+  table.mem_size = 0;
+
+  if (type == KEY)
+  {
+    key_table = table;
```
Comment (b-carr) [inline src/util/common_table.c:81]: I'm not sure I understand what this assignment, or the assignment on line 85 are doing. ```table``` is already the correct one of ```key_table``` or ```server_table``` from the switch on line 39. I don't see anything in the code that would cause ```table``` to not be the correct structure here already.

File: src/util/common_table.c [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  if (type == SERVER)
+  {
+    server_table = table;
+  }
+
+  pelz_log(LOG_DEBUG, "Table Destroy Function Complete");
+  return (0);
+}
+
+int table_delete(int type, charbuf id)
```
Comment (b-carr) [inline src/util/common_table.c:92]: Correct type of ```type```.

File: src/util/common_table.c [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    key_table = table;
+  }
+  else if (type == SERVER)
+  {
+    server_table = table;
+  }
+  return OK;
+}
+
+int table_lookup(int type, charbuf id, int *index)
```
Comment (b-carr) [inline src/util/common_table.c:181]: Correct type of ```type```.

File: src/util/pelz_io.c [hunk 9]
```diff
@@ -511,10 +512,18 @@ ParseResponseStatus parse_pipe_message(char **tokens, size_t num_tokens)
         }
 
         free(data);
+        kmyth_unsealed_data_table_initialize(eid, &ret);
```
Comment (b-carr) [inline src/util/pelz_io.c:515]: This initialization (and any associated cleanup) should be handled in ```main```.

File: src/util/pelz_request_handler.c [hunk 10]
```diff
@@ -10,15 +11,26 @@
 RequestResponseStatus pelz_request_handler(RequestType request_type, charbuf key_id, charbuf data, charbuf * output)
 {
   charbuf key;
+  int index;
 
-  if (key_table_lookup(key_id, &key))
+  if (table_lookup(KEY, key_id, &index) == 1)
```
Comment (b-carr) [inline src/util/pelz_request_handler.c:16]: Why did you add the explicit ```== 1``` here, but not in the next ```if``` statement?

**Review by b-carr — COMMENTED** [review]
I'm not all the way through poking at this code, but the current set of comments will at least give you some things to look at while I finish with it.

File: src/util/common_table.c (hunk 6 above)
Comment (JWZeta) [reply src/util/common_table.c:81]: ""table"" is a local variable and ''"key_table"" or ""server_table"" are global variables so the changes to ""table"" will not stay after the function if you don't reassign the changes back to the global variable.

File: include/server_table.h (hunk 1 above)
Comment (b-carr) [reply include/server_table.h]: I apologize that I wasn't more clear, I meant the whole file, not just that line.

File: src/util/common_table.c (hunk 6 above)
Comment (b-carr) [reply src/util/common_table.c:81]: In that case I suggest making ```table``` a pointer to one of ```key_table``` or ```server_table``` so you can avoid the re-assignment.

**Review by b-carr — APPROVED** [review]
